windows-mixin/dashboards/windows_exporter.libsonnet

```diff
@@ -0,0 +1,422 @@
+local win = import './wintable.libsonnet';
+local g = import 'grafana-builder/grafana.libsonnet';
+local grafana = import 'grafonnet/grafana.libsonnet';
+local prometheus = grafana.prometheus;
+local graphPanel = grafana.graphPanel;
+
+local host_matcher = 'job=~"$job", agent_hostname=~"$hostname"';
+
+// Templates
+local ds_template = {
+  current: {
+    text: 'default',
+    value: 'default',
+  },
+  hide: 0,
+  label: 'Data Source',
+  name: 'prometheus_datasource',
+  options: [],
+  query: 'prometheus',
+  refresh: 1,
+  regex: '',
+  type: 'datasource',
+};
+
+local job_template = grafana.template.new(
+  'job',
+  '$prometheus_datasource',
+  'label_values(windows_cs_hostname, job)',
+  label='Job',
+  refresh='load',
+  multi=true,
+  includeAll=true,
+  allValues='.+',
+  sort=1,
+);
+
+local hostname_template = grafana.template.new(
+  'hostname',
+  '$prometheus_datasource',
+  'label_values(windows_cs_hostname{job=~"$job"}, hostname)',
+  label='Hostname',
+  refresh='load',
+  multi=true,
+  allValues='.+',
+  sort=1,
+);
+
+{
+  grafanaDashboards+:: {
+    'windows_exporter.json':
+      grafana.dashboard.new(
+        'Windows Overview',
+        time_from='%s' % $._config.dashboardPeriod,
+        editable=false,
+        tags=($._config.dashboardTags),
+        timezone='%s' % $._config.dashboardTimezone,
+        refresh='%s' % $._config.dashboardRefresh,
+        uid='windows-overview',
+      )
+
+      .addTemplates([
+        ds_template,
+        job_template,
+        hostname_template,
+      ])
+
+      // Status Row
+      .addPanel(grafana.row.new(title='Integration Status'), gridPos={ x: 0, y: 0, w: 0, h: 0 })
+      // Integration status
+      .addPanel(integration_status_panel, gridPos={ x: 0, y: 0, w: 8, h: 2 })
+      // Latest metric received
+      .addPanel(latest_metric_panel, gridPos={ x: 8, y: 0, w: 8, h: 2 })
+
+      .addPanel(grafana.row.new(title='Overview'), gridPos={ x: 0, y: 2, w: 0, h: 0 })
+      .addPanel(usageTable, gridPos={ x: 0, y: 2, w: 24, h: 8 })
+
+      .addPanel(grafana.row.new(title='Overview Graphs'), gridPos={ x: 0, y: 10, w: 0, h: 0 })
+      .addPanel(perCpu, gridPos={ x: 0, y: 10, w: 12, h: 6 })
+      .addPanel(perMemory, gridPos={ x: 12, y: 10, w: 12, h: 6 })
+
+      .addPanel(grafana.row.new(title='Resource details : All'), gridPos={ x: 0, y: 16, w: 0, h: 0 })
+      .addPanel(uptime, gridPos={ x: 0, y: 16, w: 8, h: 4 })
+      .addPanel(errorService, gridPos={ x: 8, y: 16, w: 8, h: 4 })
+      .addPanel(diskUsage, gridPos={ x: 16, y: 16, w: 8, h: 4 })
+      .addPanel(diskIO, gridPos={ x: 0, y: 20, w: 8, h: 8 })
+      .addPanel(networkUsage, gridPos={ x: 8, y: 20, w: 8, h: 8 })
+      .addPanel(iisConnections, gridPos={ x: 16, y: 20, w: 8, h: 8 }),
+
+  },
+
+  local integration_status_panel =
```

Comment:
This row/panel seems to be used in many mixins. What do you think about moving it to separate lib and import it here and in other tabs?

Reply:
I remember we disussed in one of the meetings that it would be good to show just the status of the metrics being received from the agent - like if we have any data from the agent for the time period and what was the latest metric received. 

If we are happy with these panels to be used in all the mixins I can create a lib. It basically needs just one basic metric from the mixin like up or total containers etc to show the status as well as the latest metric received.

<img width="986" alt="image" src="https://user-images.githubusercontent.com/17196882/168603206-0c54fb31-7796-43dd-9c4a-66a4be415af9.png">

Reply:
Do we need them on mixin level or integration level?

Reply:
I think this needs to be on each dashboard.